contracts/blob-registry/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+#[near(contract_state, serializers=[borsh, json])]
+pub struct Contract {
+    info: HashMap<Namespace, Metadata>,
+}
+
+/// Repository information, understood as a set of namespaces and their metadata.
+#[derive(Default, Clone)]
+#[near(serializers=[borsh, json])]
+pub struct Metadata {
+    priority: Priority,
+    maintainers: HashSet<Maintainer>,
+    extra: Option<String>,
+}
+
+type Namespace = u32;
+type Priority = u32;
+type Maintainer = Vec<u8>;
+
+#[near]
+impl Contract {
+    #[init]
+    /// Create a new contract with a given owner.
+    pub fn new(owner_id: AccountId) -> Self {
+        assert!(!env::state_exists(), "{ERR_CONTRACT_INITIALIZED}");
+        let mut contract = Self {
+            info: Default::default(),
+        };
+        Owner::init(&mut contract, &owner_id);
+        contract
+    }
+
+    /// Get the priority level.
+    pub fn priority(&self, namespace: Namespace) -> Option<Priority> {
+        self.info.get(&namespace).map(|metadata| metadata.priority)
+    }
+
+    /// Get the maintainers.
+    pub fn maintainers(&self, namespace: Namespace) -> Option<HashSet<Maintainer>> {
+        self.info
+            .get(&namespace)
+            .map(|metadata| metadata.maintainers.clone())
+    }
+
+    /// Get the extra information in the metadata.
+    pub fn extra(&self, namespace: Namespace) -> Option<String> {
+        self.info
+            .get(&namespace)
+            .and_then(|metadata| metadata.extra.clone())
+    }
+
+    /// Add a new maintainer.
+    pub fn add_maintainer(&mut self, namespace: Namespace, maintainer: Maintainer) {
+        match self.check_authorized(namespace) {
+            Some(mut metadata) => {
+                // add it to the set and log the inclusion
+                if metadata.maintainers.insert(maintainer.clone()) {
+                    EventLog::maintainer(maintainer);
+                };
+            }
+            None => {
+                env::panic_str(ERR_UNAUTHORIZED_CALLER);
+            }
+        }
+    }
+
+    /// Submit the blob and the namespace.
+    pub fn submit(&self, namespace: Namespace, blob: Vec<u8>) {
+        // check the namespace exists and the caller is in the maintainers list
+        match self.check_authorized(namespace) {
+            Some(_) => {
+                // TODO: apart from logging, should there something else here?
+                env::input()
+                    .is_none()
+                    .then(|| env::panic_str(ERR_INVALID_INPUT));
+                EventLog::blob(namespace, blob);
+            }
+            None => {
+                env::panic_str(ERR_UNAUTHORIZED_CALLER);
+            }
+        }
+    }
+
+    /// Transfer the ownership of the contract.
+    pub fn transfer_ownership(&mut self, new_owner_id: AccountId) {
+        Self::require_owner();
+        assert_one_yocto();
+        Owner::update_owner(self, Some(new_owner_id.clone()));
+        EventLog::owner(self.own_get_owner(), Some(new_owner_id));
+    }
```

Comment:
Slight change to this:

```suggestion
    pub fn transfer_ownership(&mut self, new_owner_id: AccountId) {
        Self::require_owner(); << This functionality is essentially what the below does
        assert_one_yocto();
        Owner::update_owner(self, Some(new_owner_id.clone())); << This embeds above, I believe
        EventLog::owner(self.own_get_owner(), Some(new_owner_id)); << I don't think this event is needed, you can check those in the tests
    }
```